docs/artifacts.md

```diff
@@ -155,6 +155,82 @@ spec:
 
 It is recommended to use [purl format](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst) for artifacts uri as shown in the example. 
 
+### Output Artifacts in SLSA Provenance
+By default, Tekton Chains will consider all output artifacts as `byProducts` when generating in the [SLSA provenance](https://slsa.dev/spec/v1.0/provenance). In order to treat an artifact as a [subject](https://slsa.dev/spec/v1.0/provenance#schema) of the build, you must set a boolean field `"isBuildArtifact": true` for the output artifact.
+
+e.g.  
+```yaml
+apiVersion: tekton.dev/v1
+kind: TaskRun
+metadata:
+  generateName: step-artifacts-
+spec:
+  taskSpec:
+    description: |
+      A simple task that populates artifacts to TaskRun stepState
+    steps:
+      - name: artifacts-producer
+        image: bash:latest
+        script: |
+          cat > $(artifacts.path) << EOF
+          {
+            "outputs":[
+              {
+                "name":"image",
+                "isBuildArtifact": true,
+                "values":[
+                  {
+                    "uri":"pkg:oci/nginx:stable-alpine3.17-slim?repository_url=docker.io/library",
+                    "digest":{
+                      "sha256":"df85b9e3983fe2ce20ef76ad675ecf435cc99fc9350adc54fa230bae8c32ce48",
+                      "sha1":"95588b8f34c31eb7d62c92aaa4e6506639b06ef2"
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+          EOF
+```
+This informs Tekton Chains your desire to handle the artifact.
+
+When authoring a `StepAction` or a `Task`, you can even parametrize this field and let the user of the `StepAction`/`Task` indicate their desire. Generally, things like "gcs-upload" might want to set this value by default to `false`, however, a `StepAction/Task` like "kaniko" (since it is uploading images to container registries, which are normally build products) might want to set the default to `true`. 
+
+e.g.
+
+```yaml
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: artifact-producer
+spec:
+  params:
+    - name: isBuildArtifact
+      default: true
+  steps:
+    image: bash:latest
+    script: |
+      cat > $(artifacts.path) << EOF
+      {
+        "outputs":[
+          {
+            "name":"image",
+            "isBuildArtifact": $(params.isBuildArtifact),
+            "values":[
+              {
+                "uri":"pkg:oci/nginx:stable-alpine3.17-slim?repository_url=docker.io/library",
+                "digest":{
+                  "sha256":"df85b9e3983fe2ce20ef76ad675ecf435cc99fc9350adc54fa230bae8c32ce48",
+                  "sha1":"95588b8f34c31eb7d62c92aaa4e6506639b06ef2"
+                }
+              }
+            ]
+          }
+        ]
+      }
+      EOF
+```
```

Comment:
I'd remove this - this is largely the same of the example above and doesn't add much.